src/org/zaproxy/zap/extension/replacer/ReplacerAPI.java

```
@@ -0,0 +1,202 @@
+            if (!extension.getParams().setEnabled(params.getString(PARAM_DESC), this.getParam(params, PARAM_BOOL, false))) {
+                throw new ApiException(ApiException.Type.DOES_NOT_EXIST, "description");
+            }
+            try {
+                this.extension.getParams().getConfig().save();
+            } catch (ConfigurationException e) {
+                throw new ApiException(ApiException.Type.INTERNAL_ERROR);
+            }
+        } else if (ACTION_ADD_RULE.equals(name)) {
+            String desc = params.getString(PARAM_DESC);
+            MatchType type;
+            try {
+                type = MatchType.valueOf(params.getString(PARAM_MATCH_TYPE));
+            } catch (IllegalArgumentException e1) {
+                throw new ApiException(ApiException.Type.ILLEGAL_PARAMETER, PARAM_MATCH_TYPE, e1);
+            }
+            String matchString = params.getString(PARAM_MATCH_STRING);
+            boolean matchRegex = this.getParam(params, PARAM_MATCH_REGEX, false);
+            if (matchRegex) {
+                try {
+                    Pattern.compile(matchString);
+                } catch (PatternSyntaxException e) {
+                    throw new ApiException(ApiException.Type.ILLEGAL_PARAMETER, PARAM_MATCH_STRING, e);
+                }
+            }
+            List<Integer> initiators = null;
+            String initString = this.getParam(params, PARAM_INITIATORS, "");
+            if (initString.length() > 0) {
+                initiators = new ArrayList<Integer>();
+                try {
+                    for (String str : initString.split(",")) {
+                        initiators.add(Integer.parseInt(str.trim()));
+                    }
+                } catch (NumberFormatException e) {
+                    throw new ApiException(ApiException.Type.ILLEGAL_PARAMETER, PARAM_INITIATORS, e);
+                }
+            }
+
+            if (this.extension.getParams().getRule(desc) != null) {
```

This could be done earlier (immediately after obtaining it?) to fail fast/sooner.